open_spiel/games/dark_chess.cc

```diff
@@ -0,0 +1,548 @@
+// Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "open_spiel/games/dark_chess.h"
+
+#include <algorithm>
+#include <memory>
+#include <utility>
+#include <vector>
+#include <open_spiel/fog/observation_history.h>
+
+#include "open_spiel/spiel_utils.h"
+
+namespace open_spiel {
+namespace dark_chess {
+namespace {
+
+constexpr int kNumReversibleMovesToDraw = 100;
+constexpr int kNumRepetitionsToDraw = 3;
+
+// Facts about the game.
+const GameType kGameType{
+    /*short_name=*/"dark_chess",
+    /*long_name=*/"Dark Chess",
+    GameType::Dynamics::kSequential,
+    GameType::ChanceMode::kDeterministic,
+    GameType::Information::kImperfectInformation,
+    GameType::Utility::kZeroSum,
+    GameType::RewardModel::kTerminal,
+    /*max_num_players=*/2,
+    /*min_num_players=*/2,
+    /*provides_information_state_string=*/true,
+    /*provides_information_state_tensor=*/false,
+    /*provides_observation_string=*/true,
+    /*provides_observation_tensor=*/true,
+    /*parameter_specification=*/{
+      {"board_size", GameParameter(8)},
+      {"fen", GameParameter(GameParameter::Type::kString, false)}}
+};
+
+std::shared_ptr<const Game> Factory(const GameParameters &params) {
+  return std::shared_ptr<const Game>(new DarkChessGame(params));
+}
+
+REGISTER_SPIEL_GAME(kGameType, Factory)
+
+
+std::array<bool, chess::kMaxBoardSize * chess::kMaxBoardSize> ComputePrivateInfoTable(const chess::ChessBoard& board,
```

Comment:
Maybe introduce a new constant for  chess::kMaxBoardSize * chess::kMaxBoardSize, like chess::k2dMaxBoardSize, it takes too much space while reading.